### DISCUSSION TITLE
Clearing line, enter insert mode?

### DISCUSSION THREAD

**Discussion by superDuperCyberTechno** [helix-editor/helix#4225] [Q&A] [answered]
I realized that this is something I do quite often in Vim: `Vc`

To achieve this in Helix, I need to do the following: `xdO`

Is there an easier way to do this?

**Comment by CptPotato** [comment, marked as answer]
Another way to select the current line (minus trailing and leading white space) is `x_` / `X_`.

I have this bound to `Alt-x` so I can change the current line using <kbd>Alt</kbd> + <kbd>x</kbd> <kbd>c</kbd>:

```toml
A-x = [ "extend_to_line_bounds", "trim_selections" ]
```

So you could bind this to `V` and use `Vc` as you mentioned.

In theory putting it all in one key bind `[ "extend_to_line_bounds", "trim_selections", "change_selection" ]` should also be possible but it currently does not work. There's an open issue if I remember correctly but I didn't find it after a quick search.

**Reply by ChrHorn** [reply]
> [ "extend_to_line_bounds", "trim_selections", "change_selection" ]

This should work now. There used to be an issue with binding mode-switches but it was resolved.

**Comment by superDuperCyberTechno** [comment]
I thought of a theoretical alternative.

Would it be reasonable to request the following as an addition to the Match mode?

`mal` identical to `x` (maybe include prepended indentation)
`mil` identical to `x_`

The mnemonics in Helix are really good and adding these makes sense IMO.

**Reply by sudormrfbin** [reply]
~Could you open a separate issue?~ (created in https://github.com/helix-editor/helix/issues/4234). These exact keybinds have been on my mind for some time, with the difference that `mal` would select the line without the ending newline (whereas `x` selects with the new line).

Another idea would be to have:

1. `alt-x` as `x_` (which removes indentation and newline from selection) and `alt-X` as `x` without newline (keeping indentation intact).

This would require moving the `shrink_to_line_bounds` keybind from `alt-x` to some other key, but makes it easier to do things like `alt-x` `c` to mimic vim's `cc`.

2. Or we could move `shrink_to_line_bounds` from `alt-x` to `alt-X` (which pairs with `extend_to_line_bounds` that is actually on `X`) and have only `x_` bound to `alt-x`.

**Comment by rcorre** [comment]
In vim,`Vy` and `Vd` will yank/delete _including_ the newline, but `Vc` will change _excluding_ the newline. I added this binding based on @CptPotato's suggestion:

```
X = ["extend_to_line_bounds", "trim_selections"]
```

However, now I have to remember to use `Xc` to change a line, but `xy`/`xd` to yank/delete a line, if I want it to work like vim. This wasn't ideal, so I tried this instead:

```
c = ["trim_selections", "change_selection"]
```

Now `xc`, `xy`, and `xd` all parallel the behavior of `Vc`, `Vy`, and `Vd` in vim. Of course, this will effect `c` in all contexts, but so far I've liked the behavior. For example, given "( one two three )", if I have the cursor at the start of "three" and type `wcfour<esc>`, I get "( one two four )" instead of "( one two four)".
